server/src/main/java/io/spine/server/entity/Transaction.java

```diff
@@ -303,11 +322,36 @@ void rollback(Throwable cause) {
         final S currentState = currentBuilderState();
         final TransactionListener<I, E, S, B> listener = getListener();
         listener.onTransactionFailed(cause, getEntity(), currentState,
-                                          getVersion(), getLifecycleFlags());
+                                     getVersion(), getLifecycleFlags());
         this.active = false;
         entity.releaseTransaction();
     }
 
+    /**
+     * Creates an {@link EntityRecord} for the entity under transaction.
+     *
+     * <p>Since an entity transaction delegates its {@link EntityWithLifecycle#getLifecycleFlags()}
+     * method invocations to the transaction, an instance of {@link LifecycleFlags} should be
```

Comment:
This still does not explain why the lifecycle flags which are owned by this `Transaction` instance cannot be used in this instance method.

As this is a tricky place, please say more on the subject.